pkgs/dart_mcp_server/lib/src/utils/analytics.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:dart_mcp/server.dart';
+import 'package:unified_analytics/unified_analytics.dart';
+
+/// An interface class that provides a access to an [Analytics] instance, if
+/// enabled.
+///
+/// Also provides a [clientInfo] getter which is required for analytics
+/// requests.
+///
+/// The `DartMCPServer` class implements this class so that [Analytics]
+/// methods can be easily mocked during testing.
+abstract interface class AnalyticsSupport {
+  Analytics? get analytics;
+
+  ClientImplementation get clientInfo;
+}
+
+enum AnalyticsEvent { callTool, readResource }
+
+final class ReadResourceMetrics extends CustomMetrics {
+  /// The kind of resource that was read.
+  ///
+  /// We don't want to record the full URI.
+  final String kind;
+
+  /// The length of the resource.
+  final int length;
+
+  /// The time it took to read the resource.
+  final int elapsedMilliseconds;
+
+  ReadResourceMetrics({
+    required this.kind,
+    required this.length,
+    required this.elapsedMilliseconds,
+  });
+
+  @override
+  Map<String, Object> toMap() => {
+    'kind': 'runtimeErrors',
```

Comment:
should this be using the `kind` field instead of the 'runtimeErrors' string?

Reply:
yes 🤣